netci.groovy

```diff
@@ -86,6 +86,11 @@ class Constants {
                'tailcallstress'                 : ['COMPlus_TailcallStress' : '1'],
                'jitsse2only'                    : ['COMPlus_EnableAVX' : '0', 'COMPlus_EnableSSE3_4' : '0'],
                'jitnosimd'                      : ['COMPlus_FeatureSIMD' : '0'],
+               'jitnox86hwintrinsic'            : ['COMPlus_EnableIncompleteISAClass' : '1', 'COMPlus_EnableSSE' : '0' , 'COMPlus_EnableSSE2' : '0' , 'COMPlus_EnableSSE3' : '0' , 'COMPlus_EnableSSSE3' : '0' , 'COMPlus_EnableSSE41' : '0' , 'COMPlus_EnableSSE42' : '0' , 'COMPlus_EnableAVX' : '0' , 'COMPlus_EnableAVX2' : '0' , 'COMPlus_EnableAES' : '0' , 'COMPlus_EnableBMI1' : '0' , 'COMPlus_EnableBMI2' : '0' , 'COMPlus_EnableFMA' : '0' , 'COMPlus_EnableLZCNT' : '0' , 'COMPlus_ EnablePCLMULQDQ' : '0' , 'COMPlus_EnablePOPCNT' : '0'],
+               'jitx86hwintrinsic'              : ['COMPlus_EnableIncompleteISAClass' : '1'],
+               'jitx86hwintrinsicsseencoding'   : ['COMPlus_EnableIncompleteISAClass' : '1', 'COMPlus_EnableAVX' : '0'],
```

Comment:
Does disabling AVX also disable AVX2?

Also, would `LegacyEncoding` be a better name? It seems the Architecture Manual refers to them (most generally) as `VEX.128 encoded version`, `VEX.256 encoded version`, and `Legacy SSE version`

Reply:
or maybe `noavx`?

Reply:
> Does disabling AVX also disable AVX2?

Yes. `EnableAVX` was introduced by `Vector<T>`, so it should control `AVX` and `AVX2` both.

> Also, would LegacyEncoding be a better name?

No preference, both ok to me.

Reply:
Question. Can we have a shortcut for a group of tests? For example, `jitx86hwintrinsic` can trigger all these new CI jobs.

Reply:
> Can we have a shortcut for a group of tests?
Yes. I forget the exact syntax. @mmitche might know. Otherwise, I'll look it up in a bit and get back to you.

Reply:
> so it should control AVX and AVX2 both.

Does the same logic work for the others? That is, if I disable `COMPlus_EnableSSE`, does SSE2 still work, or is it disabled given the generational nature of these ISAs?

This is probably important as some people may assume the generational nature always hold true. It is also probably important for the cases where certain helper APIs (like `StaticCast`) are in the base generation, but not also in the latter.

Reply:
> Does the same logic work for the others? That is, if I disable COMPlus_EnableSSE, does SSE2 still work, 

No.

Reply:
Ok, so to confirm `COMPlus_SSE=0` will also disable `SSE2`, `SSE3`, `SSSE3`, `SSE41`, and `SSE42`, correct? Does it also control `AVX` and `AVX2`?

Do we have this documented anywhere yet?

Reply:
> Ok, so to confirm COMPlus_SSE=0 will also disable SSE2, SSE3, SSSE3, SSE41, and SSE42, correct? Does it also control AVX and AVX2?

No, `COMPlus_SSE=0` just control SSE.

Reply:
> This is probably important as some people may assume the generational nature always hold true. 

I think each environment variable should control itself (except `AVX`) because the generate-to-generation relationship is not always consistent and these knobs are just for debugging.

Reply:
> these knobs are just for debugging

These knobs will do nothing if set in a Release build of the runtime? I am checking since `AVX` and `SIMD` (for example) work in both Release and Debug builds of the runtime.

> the generate-to-generation relationship is not always consistent

When is it not consistent? An inconsistency here may cause issues for the non-ISA specific helper functions (such as `StaticCast`).

It would also be useful to have such inconsistencies documented so that users can reliably know where they need to check (ex: `if (Avx.IsSupported) { /* Can I use SSE here? */ }` or `if (Sse2.IsSupported) { /* Can I use SSE here? */ }`

Reply:
> These knobs will do nothing if set in a Release build of the runtime?

Right, they only work with debug/checked build.

> When is it not consistent?

I may need to say "unclear". For example, if `SSE` can control `SSE2`, should `SSE2` control `SSE3`/ `AES`/ `AVX`? 
Originally, most of these variables were introduced for testing `IsSupported`. A very precise order may be too complex for users.